ClemBot.Bot/bot/cogs/ping_pong_cog.py

```diff
@@ -13,32 +14,67 @@ def __init__(self, bot: ClemBot):
-        await self.bot.health_check_route.ping()
-        clembot_api_latency = time.perf_counter() - start
+    async def ping(self, ctx: ext.ClemBotCtx, url: str = "") -> None:
+        if not url:
+            start = time.perf_counter()
+            await self.bot.health_check_route.ping()
+            clembot_api_latency = time.perf_counter() - start
 
-        embed = discord.Embed(color=Colors.ClemsonOrange, title="ClemBot Latency :stopwatch:")
-        embed.add_field(name="Discord WebSocket", value=f"{self.bot.latency * 1000 : 1.2f}ms")
-        embed.add_field(name="Discord HTTP", value=":infinity:ms")
-        embed.add_field(name="ClemBot API", value=f"{clembot_api_latency * 1000 : 1.2f}ms")
-        embed.set_footer(text=str(ctx.author), icon_url=ctx.author.display_avatar.url)
+            embed = discord.Embed(color=Colors.ClemsonOrange, title="ClemBot Latency :stopwatch:")
+            embed.add_field(name="Discord WebSocket", value=f"{self.bot.latency * 1000 : 1.2f}ms")
+            embed.add_field(name="Discord HTTP", value=":infinity:ms")
+            embed.add_field(name="ClemBot API", value=f"{clembot_api_latency * 1000 : 1.2f}ms")
+            embed.set_footer(text=str(ctx.author), icon_url=ctx.author.display_avatar.url)
 
-        start = time.perf_counter()
+            start = time.perf_counter()
 
-        sent_message = await ctx.send(embed=embed)
+            sent_message = await ctx.send(embed=embed)
 
-        discord_http_latency = (time.perf_counter() - start) * 1000
+            discord_http_latency = (time.perf_counter() - start) * 1000
 
-        embed.remove_field(1)
-        embed.insert_field_at(1, name="Discord HTTP", value=f"{discord_http_latency : 1.2f}ms")
+            embed.remove_field(1)
+            embed.insert_field_at(1, name="Discord HTTP", value=f"{discord_http_latency : 1.2f}ms")
 
-        await sent_message.edit(embed=embed)
+            await sent_message.edit(embed=embed)
+        else:
+            timeout = 3
+            status = "success"
+            start = time.perf_counter()
+            try:
+                requests.get(url, timeout=timeout)
+            except requests.ConnectionError:
+                status = "conn_error"
+            except requests.ConnectTimeout:
+                status = "conn_timeout"
+            except (
+                requests.exceptions.InvalidURL,
+                requests.exceptions.MissingSchema,
+                requests.exceptions.InvalidSchema,
+            ):
+                status = "invalid"
+            elapsed = time.perf_counter() - start
+
+            messages = {
+                "success": f"The request to '{url}' returned successfully in {elapsed * 1000 : 1.2f}ms.",
+                "conn_error": f"The server at '{url}' did not respond.",
+                "conn_timeout": f"The server at '{url}' did not respond within {timeout} seconds.",
+                "invalid": f"The URL '{url}' is invalid.",
+            }
```

Comment:
We should never use strings as semantic/logical switches. stringly typed code is an anti pattern. Instead here we should just build the result string in each except handler. thats the easiest approach.